rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java

```diff
@@ -2194,6 +2195,186 @@ void save_callsStorageProviderSave() {
+
+            assertCallWasSuccessful(commitFederationCall);
+        }
+
+        @Test
+        void commitFederationAccordingToActivations_postRSKIP186_preRSKIP419_shouldCallLegacyCommitFederationAndPreserveFederationChangeInfo() {
+            // arrange
+            activations = ActivationConfigsForTest.arrowhead631().forBlock(0L);
+            federationSupport = new FederationSupportImpl(federationMainnetConstants, federationStorageProvider, rskExecutionBlock, activations);
+            List<UTXO> utxosToMove = new ArrayList<>(federationStorageProvider.getNewFederationBtcUTXOs(federationMainnetConstants.getBtcParams(), activations));
+
+            // act
+            FederationChangeResponseCode commitFederationCall = federationSupport.commitFederationAccordingToActivations(false, pendingFederation.getHash(), bridgeEventLogger);
+
+
+            // assertions
+
+            assertUTXOsWereMoved(utxosToMove);
+
+            assertNewAndOldFederationsWereSet();
+
+            assertPendingFederationVotingWasCleaned();
+
+            assertFederationChangeInfoWasSet();
+
+            Federation oldFederation = federationStorageProvider.getOldFederation(federationMainnetConstants, activations);
+            Federation newFederation = federationStorageProvider.getNewFederation(federationMainnetConstants, activations);
+            assertLogCommitFederation(oldFederation, newFederation);
+
+            assertCallWasSuccessful(commitFederationCall);
+        }
+
+        @Test
+        void commitFederationAccordingToActivations_postRSKIP419_shouldCallCommitFederation() {
+            // arrange
+            activations = ActivationConfigsForTest.all().forBlock(0L);
+            federationSupport = new FederationSupportImpl(federationMainnetConstants, federationStorageProvider, rskExecutionBlock, activations);
+            Federation activeFederation = federationSupport.getActiveFederation();
+            Federation federationBuiltFromPendingFederation =
+                pendingFederation.buildFederation(Instant.ofEpochMilli(rskExecutionBlockTimestamp), rskExecutionBlockNumber, federationMainnetConstants, activations);
+
+            // act
+            FederationChangeResponseCode commitFederationCall = federationSupport.commitFederationAccordingToActivations(false, pendingFederation.getHash(), bridgeEventLogger);
+
+
+            // assertions
+
+            // assert proposed federation was set correctly
+            Optional<Federation> proposedFederation =
+                federationStorageProvider.getProposedFederation(federationMainnetConstants, activations);
+            assertTrue(proposedFederation.isPresent());
+            assertEquals(federationBuiltFromPendingFederation, proposedFederation.get());
+
+            assertPendingFederationVotingWasCleaned();
+
+            assertFederationChangeInfoWasSet();
+
+            assertLogCommitFederation(activeFederation, proposedFederation.get());
```

Comment:
I believe we should assert that this test is really calling the new `commitFederation` method. As it is now, pre and post RSKIP419 both use the same assertions. I think we missing an assertion that allows us to differentiate which one was called. Wdyt?

Reply:
yes, i had thought about that but is not an assertion that can be added in this same test (because in that case the federationSupport should be mocked, and we don't want that).

I also think that is kind of covered since the legacyCommitFederation does not have the proposed federation part. So if the proposedFederation is set, we know that the method called was the new one.

That being said, i think that adding a new test that proves what you say with mocks will add some value.
wdyt?

Reply:
i added the test cases :)